turbinia/processors/archive.py

```diff
@@ -0,0 +1,118 @@
+    compressed_directory(str): The path to the compressed tar file.
+
+  Raises:
+    TurbiniaException: If validation fails.
+  """
+  if not os.path.exists(compressed_directory):
+    raise TurbiniaException(
+        'The File or Directory does not exist: {0:s}'.format(
+            compressed_directory))
+
+  # TODO(wyassine): rewrite this check so it is not dependant
+  # on a list of hard coded extensions and instead have a
+  # check to determine whether or not it is a tar file format.
+  if not (compressed_directory.endswith('.tgz') or
+          compressed_directory.endswith('.tar.gz')):
+    raise TurbiniaException(
+        'The file is not a supported format. The list of '
+        'acceptable exensions are: .tgz or .tar.gz')
+
+
+def CompressDirectory(uncompressed_directory):
+  """Compress a given directory into a tar file.
+
+  Args:
+    uncompressed_directory(str): The path to the uncompressed directory.
+
+  Returns:
+    str: The path to the tar file.
+  """
+  # Error handling check for a non-existent file or directory.
+  if not os.path.exists(uncompressed_directory):
+    raise TurbiniaException(
+        'The File or Directory does not exist: {0:s}'.format(
+            uncompressed_directory))
+
+  # Iterate through a given list of files and compress them.
+  compressed_directory = uncompressed_directory + '.tar.gz'
+  try:
+    with tarfile.TarFile.open(compressed_directory, 'w:gz') as tar:
+      tar.add(uncompressed_directory, arcname='')
+      tar.close()
+      log.info(
+          'The tar file has been created and '
+          'can be found at: {0:s}'.format(compressed_directory))
+  except IOError as exception:
+    raise TurbiniaException('An error has occured: {0:s}'.format(exception))
+  except tarfile.TarError as exception:
+    raise TurbiniaException(
+        'An error has while compressing the directory: {0:s}'.format(exception))
+  return compressed_directory
+
+
+def UncompressTarFile(compressed_directory, output_tmp):
+  """Uncompress a provided tar file.
+
+  Args:
+    compressed_directory(str): The path to the tar file.
+
+  Returns:
+    str: The path to the uncompressed directory.
+  """
+  # Tar file validation check
+  ValidateTarFile(compressed_directory)
+
+  # Generate the uncompressed directory path
+  uncompressed_file = 'uncompressed-' + str(int(time()))
+  uncompressed_directory = os.path.join(output_tmp, uncompressed_file)
+
+  # Uncompress the tar file into the uncompressed directory.
+  try:
+    tar = tarfile.TarFile.open(compressed_directory)
+    tar.extractall(path=uncompressed_directory)
+    tar.close()
+    log.info(
+        'The tar file has been uncompressed to the following directory: {0:s}'
+        .format(uncompressed_directory))
+  except IOError as exception:
+    raise TurbiniaException('An error has occured: {0:s}'.format(exception))
+  except tarfile.TarError as exception:
+    raise TurbiniaException(
+        'An error has occured while uncompressing the tar'
```

Comment:
Don't forget to leave a space at the end of this line so the words aren't concatenated.

Reply:
Fixed!